dicthash/dicthash.py

```diff
@@ -49,58 +45,66 @@ def _save_convert_float_to_int(x):
     return int(x * FLOAT_FACTOR)
 
 
-def _unpack_value(value, prefix='', whitelist=None, blacklist=None):
+def _unpack_value(value: Union[dict, Iterable, float, int],
+                  prefix: str ='', whitelist: Optional[List[Hashable]] =None,
+                  blacklist: Optional[List[Hashable]]=None) -> str:
     """
     Unpack values from a data structure and convert to string. Call
     the corresponding functions for dict or iterables or use simple
     string conversion for scalar variables.
 
     Parameters
     ----------
-    value : dict, iterable, scalar variable
+    value : Union[dict, Iterable, float, int]
         Value to be unpacked.
     prefix : str, optional
         Prefix to preprend to resulting string. Defaults to empty
         string.
+
+    Returns
+    -------
+    str
+        Unpacked values.
     """
 
-    try:
+    if isinstance(value, dict):
```

Comment:
why prefer `if/else` over `try/except`?

Reply:
`mypy` complains about this. I think this `try/except` block programming is exactly the kind of duck typing, that is in conflict with static type checkers. How can the type checker check the types with this construction? I think it is impossible.
To be honest, even before introducing the type checker, I found this construction a bit peculiar.

Reply:
but... but... if it quacks? ;) i don't have strong opinion here. ok, fine